src/relations/database_requires.py

```diff
@@ -1,211 +1,122 @@
+    @property
+    def host(self) -> str:
+        """MySQL cluster primary host"""
+        return self._endpoint.split(":")[0]
 
-    def _on_backend_database_created(self, event: DatabaseCreatedEvent) -> None:
-        """Handle the database created event."""
-        if not self.charm.unit.is_leader():
-            return
+    @property
+    def port(self) -> str:
+        """MySQL cluster primary port"""
+        return self._endpoint.split(":")[1]
 
-        if self.charm.app_peer_data.get(MYSQL_DATABASE_CREATED):
-            return
+    @property
+    def username(self) -> str:
+        """Admin username"""
+        return self._remote_databag["username"]
 
-        self.charm.app_peer_data[MYSQL_ROUTER_REQUIRES_DATA] = json.dumps(
-            {
-                "username": event.username,
-                "endpoints": event.endpoints,
-            }
-        )
-        self._create_application_user(event.username, event.password, event.endpoints)
+    @property
+    def password(self) -> str:
+        """Admin password"""
+        return self._remote_databag["password"]
 
-    def _on_endpoints_changed(self, event: DatabaseEndpointsChangedEvent) -> None:
-        """Handle the endpoints changed event.
+    def is_breaking(self, event):
+        """Whether relation will be broken after the current event is handled"""
+        return isinstance(event, ops.RelationBrokenEvent) and event.relation.id == self._id
 
-        Update the endpoint in the MYSQL_ROUTER_REQUIRES_DATA so that future
-        bootstrapping units will not fail.
+    @property
+    def _local_unit_databag(self) -> ops.RelationDataContent:
+        """Unit databag"""
+        return self._relation.data[self._interface.local_unit]
+
+    def set_router_id_in_unit_databag(self, router_id: str) -> None:
+        """Set router ID in unit databag.
+
+        Used by MySQL charm to remove router metadata from InnoDB cluster when a MySQL Router unit
+        departs the relation
         """
-        if not self.charm.unit.is_leader():
-            return
+        self._local_unit_databag["router_id"] = router_id
 
-        if self.charm.app_peer_data.get(MYSQL_ROUTER_REQUIRES_DATA):
-            requires_data = json.loads(self.charm.app_peer_data[MYSQL_ROUTER_REQUIRES_DATA])
 
-            requires_data["endpoints"] = event.endpoints
+class RelationEndpoint:
+    """Relation endpoint for MySQL charm"""
 
-            self.charm.app_peer_data[MYSQL_ROUTER_REQUIRES_DATA] = json.dumps(requires_data)
+    NAME = "backend-database"
+
+    def __init__(self, charm_: "charm.MySQLRouterOperatorCharm") -> None:
+        self._interface = data_interfaces.DatabaseRequires(
+            charm_,
+            relation_name=self.NAME,
+            # HACK: MySQL Router needs a user, but not a database
+            # Use the DatabaseRequires interface to get a user; disregard the database
+            database_name="_unused_mysqlrouter_database",
```

Comment:
The database name also should be updated.

Reply:
I don't see logic which will create any database for router user. see [if/else block here](https://github.com/canonical/mysql-operator/blob/main/src/relations/db_router.py#L143-L154) and [user creation logic here](https://github.com/canonical/mysql-operator/blob/main/lib/charms/mysql/v0/mysql.py#L413-L459).
please fix [k8s logic here](https://github.com/canonical/mysql-k8s-operator/blob/main/src/relations/mysql_provider.py#L169-L172), because it looks wrong.